### ISSUE TITLE
OBS records individual windows with low frame rate

### ISSUE THREAD

**Issue by FUFSoB** [hyprwm/Hyprland#5677]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 1ce21fdb3ec93b38c45b501a30b81dd379429a3e dirty (nix: fix missing git in wlroots build).
Date: Sun Apr 21 17:54:23 2024
Tag: v0.39.1-25-g1ce21fdb, commits: 4485

flags: (if any)


System Information:
System name: Linux
Node name: arch-pc-nvme
Release: 6.8.7-zen1-1-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Wed, 17 Apr 2024 15:20:00 +0000


GPU information: 
08:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Cezanne [Radeon Vega Series / Radeon Vega Mobile Series] [1002:1638] (rev c9) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Regression

### Description

OBS records individual windows with low frame rate when OBS or window are not focused.

Does not happen with 0.38.1, introduced with 0.39.0. I sadly dunno how to bisect this properly, couldn't get past errors with wlroots deps. `git reset --hard --recurse-submodules` and manually cloning did not help.

### How to reproduce

1. Open an app on workspace 1 (i used `vkcube`)
2. Open OBS on workspace 2 and select opened window (in my case named ` : `)
3. Start recording and switch to workspace 3 (I also opened terminal there)
4. Wait for 5 seconds, switch back, stop recording and watch the video

Expected output: frame rate is normal
Actual output: app does lag when OBS is unfocused

### Crash reports, logs, images, videos

Recording from OBS:

https://github.com/hyprwm/Hyprland/assets/32937947/a70ebcc7-a0ec-41c9-8b1f-8195f361fd2c

**Comment by vaxerski** [comment]
`git bisect` then please

**Comment by FUFSoB** [comment]
Yeah, I'd like to. Can you write brief how-to for dummies please?

<details> <summary>Bisect log</summary>

```sh
~/git/Hyprland main ❯ git bisect start                                                        07:25:55 PM
status: waiting for both good and bad commits
~/git/Hyprland main bisect ❯ git bisect good 360ede79d124ffdeebbe8401f1ac4bc0dbec2c91         07:26:00 PM
status: waiting for bad commit, 1 good commit known
~/git/Hyprland main bisect ❯ git bisect bad 4dc07c4378e0fad735fb0efc96d6200739d5cae1          07:26:09 PM
Bisecting: a merge base must be tested
warning: unable to rmdir 'subprojects/wlroots-hyprland': Directory not empty
[3875679755014997776e091ff8903acfb311dd2f] props: bump ver to 0.38.0
~/git/Hyprland #v0.38.0 bisect ?1 ❯ git reset --hard --recurse-submodules $(git rev-parse HEAD)
fatal: not a git repository: ../../.git/modules/wlroots
fatal: could not reset submodule index
```

</details>

**Comment by vaxerski** [comment]
first, `git reset --hard --recurse-submodules` before you do anything

0.38.1 is on a different branch.
Mark 0.38.0 as good and 0.39.0 as bad like you do.

then, after every bisect step (git bisect good/bad) run `git reset --hard --recurse-submodules`

Should work.

**Comment by FUFSoB** [comment]
Somehow that didn't work. I had to clone without submodules, init bisect and only then do `git submodule update --init --recursive`

Anyway, this seems like it's not a regression? Can repro on 0.30.0, haven't tested older versions. It somehow became more noticeable though.

Previous steps to repro seem like a different issue, not the one I wanted to report.

Actual steps to repro:
1. Open anything that continuously plays something on ws 1 (e.g. `vkcube` or `mpv`)
2. Open OBS on ws 2 and select opened window
3. Start recording, go to ws 3 and open any of those: `anki`, `thunderbird`, `mpv`, `vivaldi`, `foot` (that's what I tested, all report `xwayland: 0`)
   - Depending on the app, resulting frame rate is different???
   - Cannot repro if opened app is Xwayland (tested with `spotify`)
4. Stop moving a cursor and wait for some time
5. Go back to OBS, stop recording and review the video

**Comment by The-Briel-Deal** [comment]
Can you pull up your cpu/gpu usage with htop/radeontop? Curious if your gpu or cpu is just getting pinned and obs doesn't have high enough niceness. Try decreasing niceness in vkcube and increasing obs niceness.

This would be my best guess if it happens out of focus. It might be because its deprioritizing obs over vkcube when not focused. I would really guess if this isn't it then its a pipewire/portal/obs issue. Kind of doubt this is wm's doing.

I can't repro btw.

**Comment by FUFSoB** [comment]
I don't think CPU/iGPU usage is bad in my case. It instantly drops when windows are unfocused and cursor doesn't move. Disabling the fancy and manipulating with niceness does nothing (both sides).

Tried downgrading OBS, ffmpeg and pipewire to pre-april versions (last I knew were working correctly), it did not change a thing. I don't think there were any major updates to portal.

I think my config is good since this also does happen on almost default config. And in the end this doesn't look like a Hyprland regression either.

I'm out of ideas. Here's recording of both my screen and vkcube.

https://github.com/hyprwm/Hyprland/assets/32937947/5cf5acc6-8b90-49e7-b95e-1826e8d84fe2

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
